src/Bicep.LangServer/Handlers/BicepExternalSourceRequestHandler.cs

```diff
@@ -53,29 +53,22 @@ public Task<BicepExternalSourceResponse> Handle(BicepExternalSourceParams reques
             if (!moduleDispatcher.TryGetArtifactReference(ArtifactType.Module, request.Target, new Uri("file:///no-parent-file-is-available.bicep")).IsSuccess(out var moduleReference))
             {
                 telemetryProvider.PostEvent(ExternalSourceRequestFailure(nameof(moduleDispatcher.TryGetArtifactReference)));
-                throw new InvalidOperationException(
-                    $"The client specified an invalid module reference '{request.Target}'.");
+                return Task.FromResult(new BicepExternalSourceResponse(null,
+                    $"The client specified an invalid module reference '{request.Target}'."));
             }
 
             if (!moduleReference.IsExternal)
             {
                 telemetryProvider.PostEvent(ExternalSourceRequestFailure("localNotSupported"));
-                throw new InvalidOperationException(
-                    $"The specified module reference '{request.Target}' refers to a local module which is not supported by {BicepExternalSourceLspMethodName} requests.");
-            }
-
-            if (this.moduleDispatcher.GetArtifactRestoreStatus(moduleReference, out _) != ArtifactRestoreStatus.Succeeded)
```

Comment:
Instead of giving an error now when a module we're trying to show source from has not been restored (it used to only happen in that scenario), now we'll just display the compiled main.json file instead.  This can now happen in a nested module that couldn't be restored.